### DISCUSSION TITLE
Enable looking up a type placed in a collection

### DISCUSSION THREAD

**Discussion by ivevasiljevic** [LemonAppDev/konsist#1283] [Request a Feature]
### Tell us more.

I couldn't figure out what title to put to maybe present the idea. Nonetheless, for example, if in Android I have an Api/Service interface and I want to make sure so that only DTO classes are used as return types. There are some cases where I have to use a List for this case, but then I want to check if the type what has been placed in a List as a generic is actually a DTO class as well.

List<"SomeTypeDto">

If possible, it would be great if I could get the reference of this type as well. Now, I only found the baseSourceType and sourceType properties, but they are useless in this case.

Thank you

**Comment by nataliapeterwas** [comment]
Konsist `0.17.0` has been released! This release includes this improvement :) 

The following test should be helpful in your case:
```kotlin
Konsist
   .scopeFromProject()
   .functions()
   .returnTypes
   .assertTrue {
      it.hasNameEndingWith("Dto") || it.hasTypeArgument { typeArgument -> typeArgument.hasNameEndingWith("Dto") } == true
   }
```

It's also possible to get the reference of this type - see **Source Declaration (⚠️Breaking Change)** section in [Release Notes](https://github.com/LemonAppDev/konsist/releases/tag/v0.17.0)

I closed this discussion. If anything is unclear or missing, feel free to start a new one.
